### DISCUSSION TITLE
No Zookeeper configuration in server config

### DISCUSSION THREAD

**Discussion by kfeofantov** [amacneil/dbmate#481] [Support]
**Description**
Clickhouse installed with Keeper (https://clickhouse.com/docs/en/guides/sre/keeper/clickhouse-keeper). Keeper is default configuration and used instead of ZooKeeper. But **dbmate** tool fails with error: **Error: code: 139, message: There is no Zookeeper configuration in server config**

- Version: < 2.6.0
- Database: ClickHouse
- Operating System: Linux

**Steps To Reproduce**
Install Clickhouse (https://clickhouse.com/docs/en/install#quick-install) and run migration

**Expected Behavior**
Migrations up with success

**Comment by dossy** [comment]
This may be a misconfiguration of your ClickHouse installation.

ClickHouse Keeper takes the place of ZooKeeper, so the error "There is no Zookeeper configuration in server config" indicates that neither ClickHouse Keeper nor ZooKeeper are installed and configured in the server correctly.

Even though you are using ClickHouse Keeper, you still need a `zookeeper` node in your `config.xml` to tell ClickHouse where the ZooKeeper or ClickHouse Keeper nodes are:

https://clickhouse.com/docs/en/guides/sre/keeper/clickhouse-keeper#clickhouse-keeper-user-guide

> 4. Enable the Zookeeper component. It will use the ClickHouse Keeper engine:

Even though you're using ClickHouse Keeper, you still configure ClickHouse's Zookeeper config settings to connect to ClickHouse Keeper.

This isn't a dbmate-specific issue.
